### ISSUE TITLE
UWP app crash in release when adding a `TeachingTip` control

### ISSUE THREAD

**Issue by vgromfeld** [microsoft/microsoft-ui-xaml#5043]
**Describe the bug**
After adding a `TeachingTip` control in my UWP application, it no longer starts when compiled in **release** mode (.Net Native). It crashes during the app initialization (when the main page content is set). 
Everything is working well in **debug** builds.

Looking at the dump, the crash occurs during the measure pass and is triggered by the `NavigationViewItemBase` measure.

**Steps to reproduce the bug**
I haven't been able to create a small repro app yet 😓.

**Expected behavior**
The application works in release.

**Version Info**
NuGet package version: [Microsoft.UI.Xaml 2.6.0-prerelease.210415001]

**Additional context**
I've been able to workaround the issue by adding the following style in my application's resources:
```xml
<Style TargetType="muxc:IconSource">
    <Setter Property="Foreground" Value="{ThemeResource SystemControlPageTextBaseHighBrush}" />
</Style>
```

Looking at the `IconSource.Foreground` [documentation ](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.iconsource.foreground?view=winui-3.0#Microsoft_UI_Xaml_Controls_IconSource_Foreground) and [implementation](https://github.com/microsoft/microsoft-ui-xaml/blob/9052972906c8a0a1b6cb5d5c61b27d6d27cd7f11/dev/Generated/IconSource.properties.cpp#L23), we can see that the default value is `null`.

My first guess was that this `null` default value was incorrect and was preventing all the controls using an `IconSource` in their default template to load. But, if I set the `IconSource.Foreground` property to `null` in my application's resources, everything works 😲:
```xml
<Style TargetType="muxc:IconSource">
    <Setter Property="Foreground" Value="{x:Null}" />
</Style>
```

Do you have any idea about what causes this issue ?

**Comment by StephenLPeters** [comment]
@vgromfeld do you know if this worked with the 2.5 nuget package?

**Comment by StephenLPeters** [comment]
@ranjeshj we should look at this for 2.6

**Comment by vgromfeld** [comment]
@StephenLPeters It is working with the last 2.5 prerelease package

**Comment by Richasy** [comment]
Same issue. I only keep one `NavigationView`, and the application will still crash in **Release** mode. WinDbg returns the following information:

```
Failed to assign to property 'Microsoft.UI.Xaml.Controls.IconSource.Foreground'.
```

According to the type of `IconSource`, this should be an exception in the assignment of the icon inside the control.

In 2.5 it works fine, in 2.6, if I set **ControlResourceVersion** to `Version1`, it also works fine.

---

@vgromfeld workaround can solve my problem, but I still expect this problem to be fixed in 2.6.
